lib/ClangImporter/ClangImporter.cpp

```diff
@@ -6584,6 +6584,35 @@ void ClangImporter::withSymbolicFeatureEnabled(
       oldImportSymbolicCXXDecls.get());
 }
 
+const clang::TypedefType *ClangImporter::getTypeDefForCXXCFOptionsDefinition(
+    const clang::Decl *candidateDecl) {
+
+  if (!Impl.SwiftContext.LangOpts.EnableCXXInterop)
+    return nullptr;
+
+  auto enumDecl = dyn_cast<clang::EnumDecl>(candidateDecl);
+  if (!enumDecl)
+    return nullptr;
+
+  if (!enumDecl->getDeclName().isEmpty())
+    return nullptr;
+
+  clang::SourceManager &SM = getClangASTContext().getSourceManager();
+  clang::SourceLocation SpellingLoc =
+      SM.getSpellingLoc(enumDecl->getLocation());
+  clang::PresumedLoc PLoc = SM.getPresumedLoc(SpellingLoc);
+  if (PLoc.isValid() && !strstr(PLoc.getFilename(), "CFAvailability"))
+    return nullptr;
```

Comment:
If I understand correctly, this is checking if the token of `enumDecl` is defined in `CFAvailability`? (no extension?)

Not a bad test for safety, but I think it is not necessary. There are some problems with checking for this that I am not a fan of:

Apple SDK files are detached from the Swift compiler, so changes can happen in them and the compiler would broke very subtly.

What happens with macros that reproduce the same setup? We have several of those in different tests because the Apple SDK does not have the right macro yet. Those will be skipped by this check, wouldn't they? (your test used the right name, which "hides" this problem)

Let's see if someone else have an opinion about this particular bit. Not completely convinced myself that the reasons above are good enough.

Reply:
I'm sort of on the fence as well.  

I made the change because it seemed to me, we've been doing this check everywhere for "is this an anonymous enum backed by an unavailable typedef" when really we mean "is this the C++ definition of CF_OPTIONS". For example, in this case we want CF_OPTIONS to mangle the same way in C++ and Objective-C, but I'm not sure its true we want these two constructs to mangle the same in general:

```objective-c
enum Foo {};

typedef int Foo __attribute__((availability(swift, unavailable)));
enum : Foo {};
```

Seems like these are different things, they ought to behave differently, its just to keep the SDK consistent, we are forced to conflate them for CF_OPTIONS, but do we want that in general?

Fair enough on the SDK fragility, and you are right that the test file needs to be called CFAvailability.